docs/source/crud_events.rst

```diff
@@ -0,0 +1,180 @@
+**action**
+    Exact name of Nefertari view method that processes the request (action)
+
+Complete list of actions:
+    * **index** - Collection GET
+    * **create** - Collection POST
+    * **update_many** - Collection PATCH/PUT
+    * **delete_many** - Collection DELETE
+    * **collection_options** - Collection OPTIONS
+    * **show** - Item GET
+    * **update** - Item PATCH
+    * **replace** - Item PUT
+    * **delete** - Item DELETE
+    * **item_options** - Item OPTIONS
+
+And one special action:
+    * **set** - includes actions ``create``, ``update``, ``replace``, ``update_many``
+
+
+E.g. following connects ``lower_strip_processor`` subscriber to ``after_index`` event and the subscriber will be run after collection GET request is processed.
+
+.. code-block:: json
+
+    "_event_handlers": {
+        "after_index": ["lower_strip_processor"]
+    }
+
+
+We will use following subscriber to demo how to connect subscribers to events. This subscriber logs request.body.
+
+.. code-block:: python
+
+    import logging
+    log = logging.getLogger(__name__)
+
+    @registry.add
+    def log_request(event):
+        log.debug(event.request.body)
+
+
+Per-model subscribers
+---------------------
+
+To set up subscribers per-model, define ``_event_handlers`` param at the root of your model's JSON schema. In example if we have JSON schema for model ``User`` and we want to log all collection GET request to ``User`` model after they were processed (using ``log_request`` subscriber), we connect subscriber in json schema like so:
+
+
+.. code-block:: json
+
+    {
+        "type": "object",
+        "title": "User schema",
+        "$schema": "http://json-schema.org/draft-04/schema",
+        "_event_handlers": {
+            "after_index": ["log_request"]
+        },
+        ...
+    }
+
+Per-modelfield subscribers
+--------------------------
+
+To set up subscribers per-modelfield, define ``_event_handlers`` param in JSON schema of model field you want to set up subscriber for(at the same level with ``_db_settings``). Following JSON schema example connects ``after_create`` for field ``User.username``. When connected this way, subscriber will only run when ``User`` collection POST request finishes and request body contains ``username`` field.
+
+
+.. code-block:: json
+
+    {
+        "type": "object",
+        "title": "User schema",
+        "$schema": "http://json-schema.org/draft-04/schema",
+        "properties": {
+            "username": {
+                "_db_settings": {...},
+                "_event_handlers": {
+                    "after_create": ["log_request"]
+                }
+            }
+        }
+        ...
+    }
+
+
+
+Other Things You Can Do
+-----------------------
+
+You can update another field's value, for example, increment a counter. E.g. in subscriber connected to item enpoint:
+
+.. code-block:: python
+
+    @registry.add
+    def increment_count(event):
```

Comment:
this example also doesn't make sense. you would increment a counter directly, I would think, rather than setting an input parameter.